fireant/queries/builder.py

```diff
@@ -297,16 +301,49 @@ def sql(self):
         Serializes this query builder as a set of SQL queries.  This method will always return a list of one query since
         only one query is required to retrieve dimension choices.
 
-        This function only handles dimensions (select+group by) and filtering (where/having), which is everything needed
-        for the query to fetch choices for dimensions.
-
         The slicer query extends this with metrics, references, and totals.
         """
+
+        def in_hint_table(dim):
```

Comment:
this function is way too complex. Please reorganize the logic and break it up into smaller chunks